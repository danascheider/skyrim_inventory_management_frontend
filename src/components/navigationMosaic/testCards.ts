import { type MosaicCardAttributes } from './types'
import {
  YELLOW,
  PINK,
  BLUE,
  GREEN,
  AQUA
} from '../../utils/colorSchemes'

const testCards: MosaicCardAttributes[] = [
  {
    colorScheme: YELLOW,
    href: '/yellow',
    key: 'yellow',
    children: 'Yellow Card'
  },
  {
    colorScheme: PINK,
    href: '/pink',
    key: 'pink',
    children: 'Pink Card'
  },
  {
    colorScheme: BLUE,
    href: '/blue',
    key: 'blue',
    children: 'Blue Card'
  },
  {
    colorScheme: GREEN,
    href: '/green',
    key: 'green',
    children: 'Green Card'
  },
  {
    colorScheme: AQUA,
    href: '/aqua',
    key: 'aqua',
    children: 'Aqua Card'
  }
]

export default testCards