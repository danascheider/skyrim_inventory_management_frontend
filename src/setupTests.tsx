import { type ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { render } from '@testing-library/react'

export const renderWithRouter = (ui: ReactElement) => render(<BrowserRouter>{ui}</BrowserRouter>)