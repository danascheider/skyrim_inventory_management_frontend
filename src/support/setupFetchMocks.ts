import { vi } from 'vitest'
import createFetchMock from 'vitest-fetch-mock'

/*
 *
 * API Mock Setup
 *
 */

const fetchMocker = createFetchMock(vi)

fetchMocker.enableMocks()
