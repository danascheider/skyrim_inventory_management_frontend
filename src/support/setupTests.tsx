import { type ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import { LoginContext } from '../contexts/loginContext'
import { User } from 'firebase/auth'
import testProfileImg from './testProfileImg.png'

export const requireLogin = () => { /* noop */ }

export const testUser = {
  uid: 'edna',
  displayName: 'Edna St. Vincent Millay',
  email: 'edna@gmail.com',
  photoURL: testProfileImg,
  emailVerified: true,
  getIdToken: () => new Promise<string>((resolve, _reject) => resolve('xxxxxxx'))
} as User

export const renderWithRouter = (ui: ReactElement) =>
  render(<BrowserRouter>{ui}</BrowserRouter>)

export const renderAuthenticated = (
  ui: ReactElement,
  authLoading = false,
  authError = undefined
) =>
  renderWithRouter(
    <LoginContext.Provider value={{ user: testUser, token: 'xxxxxxx', requireLogin, authLoading, authError }}>
      {ui}
    </LoginContext.Provider>
  )

export const renderUnauthenticated = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider value={{ user: null, token: null, authLoading: false, requireLogin }}>
      {ui}
    </LoginContext.Provider>
  )

export const renderAuthLoading = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider value={{ user: null, token: null, authLoading: true, requireLogin }}>
      {ui}
    </LoginContext.Provider>
  )
