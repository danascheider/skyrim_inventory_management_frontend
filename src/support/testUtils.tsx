import { type ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { JSDOM } from 'jsdom'
import { render } from '@testing-library/react'
import { LoginContext } from '../contexts/loginContext'
import { User } from 'firebase/auth'
import testProfileImg from './testProfileImg.png'

export const BASE_APP_URI = 'http://localhost:5173'

declare global {
  namespace NodeJS {
    interface Global {
      document: Document
      window: Window
    }
  }
}

const setDom = () => {
  const dom = new JSDOM('<!doctype html><html><body></body></html>', {
    url: 'http://localhost:5173',
  })

  global.window = dom.window as unknown as Window & typeof globalThis
  global.document = dom.window.document
}

/*
 *
 * Global Test Data
 *
 */

export const requireLogin = () => {
  /* noop */
}

export const testUser = {
  uid: 'edna',
  displayName: 'Edna St. Vincent Millay',
  email: 'edna@gmail.com',
  photoURL: testProfileImg,
  emailVerified: true,
  getIdtoken: () =>
    new Promise<string>((resolve, _reject) => resolve('xxxxxxx')),
} as unknown as User

/*
 *
 * Test Renderers
 *
 */

const renderWithJSDOM = (ui: ReactElement) => {
  setDom()

  return render(ui)
}

export const renderWithRouter = (ui: ReactElement) =>
  renderWithJSDOM(<BrowserRouter>{ui}</BrowserRouter>)

export const renderAuthenticated = (ui: ReactElement, authLoading = false) =>
  renderWithRouter(
    <LoginContext.Provider
      value={{
        user: testUser,
        token: 'xxxxxxx',
        requireLogin,
        authLoading,
      }}
    >
      {ui}
    </LoginContext.Provider>
  )

export const renderUnauthenticated = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider
      value={{
        user: null,
        token: null,
        authLoading: false,
        requireLogin,
      }}
    >
      {ui}
    </LoginContext.Provider>
  )

export const renderAuthLoading = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider
      value={{ user: null, token: null, authLoading: true, requireLogin }}
    >
      {ui}
    </LoginContext.Provider>
  )
