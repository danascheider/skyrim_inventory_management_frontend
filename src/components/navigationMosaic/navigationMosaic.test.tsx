import { describe, test, expect } from 'vitest'
import { screen } from '@testing-library/react'
import { renderWithRouter } from '../../support/setupTests'
import testCards from './testCards'
import NavigationMosaic from './navigationMosaic'

describe('NavigationMosaic component', () => {
  test('NavigationMosaic component functions properly', () => {
    const wrapper = renderWithRouter(<NavigationMosaic cardArray={testCards} />)
    expect(wrapper).toBeTruthy()

    expect(screen.getByText('Yellow Card')).toHaveProperty('href', '/yellow')
    expect(screen.getByText('Pink Card')).toHaveProperty('href', '/pink')
    expect(screen.getByText('Blue Card')).toHaveProperty('href', '/blue')
    expect(screen.getByText('Green Card')).toHaveProperty('href', '/green')
    expect(screen.getByText('Aqua Card')).toHaveProperty('href', '/aqua')
  })

  test('matches snapshot', () => {
    const wrapper = renderWithRouter(<NavigationMosaic cardArray={testCards} />)
    expect(wrapper).toMatchSnapshot()
  })
})
