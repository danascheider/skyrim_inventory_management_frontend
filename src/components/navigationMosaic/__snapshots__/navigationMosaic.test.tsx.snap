// Vitest Snapshot v1

exports[`NavigationMosaic component > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_dcf87f"
      >
        <div
          class="_card_dcf87f"
        >
          <a
            class="_root_396ea1"
            href="/yellow"
            style="--background-color: #ffbf00; --hover-color: #e5ab00; --text-color: #4c3900;"
          >
            Yellow Card
          </a>
        </div>
        <div
          class="_card_dcf87f"
        >
          <a
            class="_root_396ea1"
            href="/pink"
            style="--background-color: #e83f6f; --hover-color: #d03863; --text-color: #fff;"
          >
            Pink Card
          </a>
        </div>
        <div
          class="_card_dcf87f"
        >
          <a
            class="_root_396ea1"
            href="/blue"
            style="--background-color: #2274a5; --hover-color: #1e6894; --text-color: #fff;"
          >
            Blue Card
          </a>
        </div>
        <div
          class="_card_dcf87f"
        >
          <a
            class="_root_396ea1"
            href="/green"
            style="--background-color: #00a323; --hover-color: #00921f; --text-color: #fff;"
          >
            Green Card
          </a>
        </div>
        <div
          class="_card_dcf87f"
        >
          <a
            class="_root_396ea1"
            href="/aqua"
            style="--background-color: #20e2e9; --hover-color: #11cbd1; --text-color: #094345;"
          >
            Aqua Card
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_dcf87f"
    >
      <div
        class="_card_dcf87f"
      >
        <a
          class="_root_396ea1"
          href="/yellow"
          style="--background-color: #ffbf00; --hover-color: #e5ab00; --text-color: #4c3900;"
        >
          Yellow Card
        </a>
      </div>
      <div
        class="_card_dcf87f"
      >
        <a
          class="_root_396ea1"
          href="/pink"
          style="--background-color: #e83f6f; --hover-color: #d03863; --text-color: #fff;"
        >
          Pink Card
        </a>
      </div>
      <div
        class="_card_dcf87f"
      >
        <a
          class="_root_396ea1"
          href="/blue"
          style="--background-color: #2274a5; --hover-color: #1e6894; --text-color: #fff;"
        >
          Blue Card
        </a>
      </div>
      <div
        class="_card_dcf87f"
      >
        <a
          class="_root_396ea1"
          href="/green"
          style="--background-color: #00a323; --hover-color: #00921f; --text-color: #fff;"
        >
          Green Card
        </a>
      </div>
      <div
        class="_card_dcf87f"
      >
        <a
          class="_root_396ea1"
          href="/aqua"
          style="--background-color: #20e2e9; --hover-color: #11cbd1; --text-color: #094345;"
        >
          Aqua Card
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
